src/nvim/mouse.c

```diff
@@ -1756,18 +1756,26 @@ static win_T *mouse_find_grid_win(int *gridp, int *rowp, int *colp)
     }
   } else if (*gridp == 0) {
     ScreenGrid *grid = ui_comp_mouse_focus(*rowp, *colp);
-    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
-      if (&wp->w_grid_alloc != grid) {
-        continue;
-      }
+    if (grid == &pum_grid) {
       *gridp = grid->handle;
-      *rowp -= grid->comp_row + wp->w_grid.row_offset;
-      *colp -= grid->comp_col + wp->w_grid.col_offset;
-      return wp;
+      *rowp -= grid->comp_row + grid->row_offset;
+      *colp -= grid->comp_col + grid->col_offset;
+      // The popup menu doesn't have a window, so return NULL
+      return NULL;
+    } else {
+      FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+        if (&wp->w_grid_alloc != grid) {
+          continue;
+        }
+        *gridp = grid->handle;
+        *rowp -= wp->w_winrow + wp->w_grid.row_offset;
+        *colp -= wp->w_wincol + wp->w_grid.col_offset;
+        return wp;
+      }
     }
 
-    // no float found, click on the default grid
-    // TODO(bfredl): grid can be &pum_grid, allow select pum items by mouse?
+    // No grid found, return the default grid, when using multigrid. With multigrid this happens for
```

Comment:
Why add "when using multigrid"? This happens without multigrid as well.

Reply:
Because that's the non-obvious case, for non-multigrid it happens for everything that isn't a float. It's probably best to remove the comment altogether.

Reply:
Ah sorry, you are referring to the extra "when using multigrid". That's not supposed to be there. 

The comment is supposed to be the following
"No grid found, return the default grid. With multigrid this happens for split separators for example."

Reply:
Yes, but the next sentence already starts with "With multigrid".

Reply:
I think you missed my previous comment. Is the comment OK if I change it to:
"No grid found, return the default grid. With multigrid this happens for split separators for example."

Reply:
Yes.